File: Sssl/Attributes.cs
using System;

namespace Sssl
{
    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
    public sealed class NotSsslFieldAttribute : Attribute { }

    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
    public sealed class SsslFieldAttribute : Attribute { }

    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
    public sealed class SsslNameAttribute : Attribute
    {
        public string Name { get; set; }

        public SsslNameAttribute(string name)
        {
            Name = name;
        }
    }
}

File: SsslFSharp/Attributes.fs
namespace SsslFSharp
open System

[<AllowNullLiteral>]
[<AttributeUsage(AttributeTargets.Property ||| AttributeTargets.Field)>]
type NotSsslFieldAttribute() = inherit Attribute()


[<AllowNullLiteral>]
[<AttributeUsage(AttributeTargets.Property ||| AttributeTargets.Field)>]
type SsslFieldAttribute() = inherit Attribute()


[<AllowNullLiteral>]
[<AttributeUsage(AttributeTargets.Property ||| AttributeTargets.Field)>]
type SsslNameAttribute(name: string) =
    inherit Attribute()
    member _.Name = name
